datafusion/expr/src/window_frame.rs

```diff
@@ -30,21 +30,24 @@ use sqlparser::ast;
 use sqlparser::parser::ParserError::ParserError;
 use std::convert::{From, TryFrom};
 use std::fmt;
+use std::fmt::Formatter;
 use std::hash::Hash;
 
 /// The frame-spec determines which output rows are read by an aggregate window function.
 ///
 /// The ending frame boundary can be omitted (if the BETWEEN and AND keywords that surround the
 /// starting frame boundary are also omitted), in which case the ending frame boundary defaults to
 /// CURRENT ROW.
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[derive(Clone, PartialEq, Eq, Hash)]
 pub struct WindowFrame {
     /// A frame type - either ROWS, RANGE or GROUPS
     pub units: WindowFrameUnits,
     /// A starting frame boundary
     pub start_bound: WindowFrameBound,
     /// An ending frame boundary
     pub end_bound: WindowFrameBound,
+    /// Flag indicates whether window frame is causal.
```

Comment:
Can you please document what `causal` means in this context (I am not sure I understand what it means)? Perhaps you can update the doc comments on `WindowFrame` to explain what the concept is and give an example of a causal bound and non causal bound and then link to that definition when referred to elsewhere?